gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -242,134 +349,362 @@ static void
+ *         If not return new pad after creation.
  */
-static gboolean
-gst_tensor_if_sink_event (GstBaseTransform * trans, GstEvent * event)
+static GstTensorPad *
+gst_tensor_if_get_tensor_pad (GstTensorIf * tensor_if,
+    gboolean * created, gint nth)
 {
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
+  GSList *walk;
+  GstPad *pad;
+  GstTensorPad *tensorpad;
+  gchar *name;
+  GstEvent *event;
+  gchar *stream_id;
+  GstCaps *caps;
+  GstTensorConfig config;
+
+  walk = tensor_if->srcpads;
+  while (walk) {
+    GstTensorPad *pad = (GstTensorPad *) walk->data;
+    if (nth == pad->nth) {
+      if (created) {
+        *created = FALSE;
+      }
+      return pad;
+    }
+    walk = walk->next;
+  }
 
-  switch (GST_EVENT_TYPE (event)) {
-    /** @todo NYI! */
-    default:
-      break;
+  tensorpad = g_new0 (GstTensorPad, 1);
+  g_assert (tensorpad != NULL);
+  GST_DEBUG_OBJECT (tensor_if, "createing pad: %d(%dth)",
+      tensor_if->num_srcpads, nth);
+
+  name = g_strdup_printf ("src_%u", tensor_if->num_srcpads);
+  pad = gst_pad_new_from_static_template (&src_factory, name);
+  g_free (name);
+
+  tensorpad->pad = pad;
+  tensorpad->nth = nth;
+  tensorpad->last_ret = GST_FLOW_OK;
+  tensorpad->last_ts = GST_CLOCK_TIME_NONE;
+
+  tensor_if->srcpads = g_slist_append (tensor_if->srcpads, tensorpad);
+  gst_tensor_if_get_tensor_config (tensor_if, &config, nth);
+
+  tensor_if->num_srcpads++;
+
+  gst_pad_use_fixed_caps (pad);
+  gst_pad_set_active (pad, TRUE);
+
+
+  if (!tensor_if->have_group_id) {
+    event =
+        gst_pad_get_sticky_event (tensor_if->sinkpad, GST_EVENT_STREAM_START,
+        0);
+    if (event) {
+      tensor_if->have_group_id =
+          gst_event_parse_group_id (event, &tensor_if->group_id);
+      gst_event_unref (event);
+    } else if (!tensor_if->have_group_id) {
+      tensor_if->have_group_id = TRUE;
+      tensor_if->group_id = gst_util_group_id_next ();
+    }
   }
 
-  /** other events are handled in the default event handler */
-  return GST_BASE_TRANSFORM_CLASS (parent_class)->sink_event (trans, event);
-}
+  stream_id =
+      gst_pad_create_stream_id (pad, GST_ELEMENT_CAST (tensor_if),
+      "other/tensors");
 
-/**
- * @brief Called when the element starts processing. optional vmethod of BaseTransform
- * @param trans "this" pointer
- * @return TRUE if there is no error.
- */
-static gboolean
-gst_tensor_if_start (GstBaseTransform * trans)
-{
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
+  event = gst_event_new_stream_start (stream_id);
+  if (tensor_if->have_group_id)
+    gst_event_set_group_id (event, tensor_if->group_id);
 
-  return FALSE; /** @todo NYI! Do not allow to start! */
+  gst_pad_store_sticky_event (pad, event);
+  g_free (stream_id);
+  gst_event_unref (event);
+
+  caps = gst_tensor_caps_from_config (&config);
+  gst_pad_set_caps (pad, caps);
+  gst_element_add_pad (GST_ELEMENT_CAST (tensor_if), pad);
```

Comment:
Is this pad activated? ( https://developer.gnome.org/gstreamer/stable/GstElement.html#gst-element-add-pad )

Reply:
activated at line 544.

Reply:
Another question: don't you need to determine the number of srcpads at the negotiation phase? It appears that this function is called while the pipeline is running.

Reply:
Considering that the else-if option is added later, So I think this is right that this function is called while the pipeline is running.

Reply:
Assuming that this will be handled later... LGTM